tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -571,127 +504,127 @@ def tridiag_forward(
 
-@pytest.mark.parametrize("left,right", [(2.0, 3.0), (3.0, 2.0)])
-def test_ternary_operator(left, right, fieldview_backend):
-    a_I_float = gtx.np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
-    b_I_float = gtx.np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
-    out_I_float = gtx.np_as_located_field(IDim)(np.zeros((size,), dtype=float64))
 
-    @gtx.field_operator(backend=fieldview_backend)
-    def ternary_field_op(
-        a: gtx.Field[[IDim], float], b: gtx.Field[[IDim], float], left: float, right: float
-    ) -> gtx.Field[[IDim], float]:
+@pytest.mark.parametrize("left, right", [(2, 3), (3, 2)])
+def test_ternary_operator(cartesian_case, left, right):
+    @gtx.field_operator
+    def testee(a: cases.IField, b: cases.IField, left: int64, right: int64) -> cases.IField:
         return a if left < right else b
 
-    ternary_field_op(a_I_float, b_I_float, left, right, out=out_I_float, offset_provider={})
-    e = np.asarray(a_I_float) if left < right else np.asarray(b_I_float)
-    assert np.allclose(e, out_I_float)
+    a = cases.allocate(cartesian_case, testee, "a")()
+    b = cases.allocate(cartesian_case, testee, "b")()
+    out = cases.allocate(cartesian_case, testee, cases.RETURN)()
 
-    @gtx.field_operator(backend=fieldview_backend)
-    def ternary_field_op_scalars(left: float, right: float) -> gtx.Field[[IDim], float]:
-        return broadcast(3.0, (IDim,)) if left > right else broadcast(4.0, (IDim,))
+    cases.verify(cartesian_case, testee, a, b, left, right, out=out, ref=(a if left < right else b))
 
-    ternary_field_op_scalars(left, right, out=out_I_float, offset_provider={})
-    e = np.full(e.shape, 3.0) if left > right else np.full(e.shape, 4.0)
-    assert np.allclose(e, out_I_float)
+    @gtx.field_operator
+    def testee(left: int64, right: int64) -> cases.IField:
+        return broadcast(3, (IDim,)) if left > right else broadcast(4, (IDim,))
 
+    e = np.asarray(a) if left < right else np.asarray(b)
+    cases.verify(
+        cartesian_case,
+        testee,
+        left,
+        right,
+        out=out,
+        ref=(np.full(e.shape, 3) if left > right else np.full(e.shape, 4)),
+    )
 
-@pytest.mark.parametrize("left,right", [(2.0, 3.0), (3.0, 2.0)])
-def test_ternary_operator_tuple(left, right, fieldview_backend):
-    a_I_float = gtx.np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
-    b_I_float = gtx.np_as_located_field(IDim)(np.random.randn(size).astype("float64"))
-    out_I_float = gtx.np_as_located_field(IDim)(np.zeros((size,), dtype=float64))
-    out_I_float_1 = gtx.np_as_located_field(IDim)(np.zeros((size,), dtype=float64))
 
-    @gtx.field_operator(backend=fieldview_backend)
-    def ternary_field_op(
-        a: gtx.Field[[IDim], float], b: gtx.Field[[IDim], float], left: float, right: float
-    ) -> tuple[gtx.Field[[IDim], float], gtx.Field[[IDim], float]]:
+def test_ternary_operator_tuple(cartesian_case):
```

Comment:
because we already do that in other places and I wanted an example of a `cases.verify` with scalars directly set inside the test. It is not a big change so I can put it back if you prefer it that way

Reply:
The problem with dropping it is that now only one of the two cases is tested. Either put it back or create another test method for the case that is not tested now.